migrations/outputs_amqp/migration.go

```diff
@@ -0,0 +1,94 @@
+package outputs_amqp
+
+import (
+	"github.com/influxdata/toml"
+	"github.com/influxdata/toml/ast"
+
+	"github.com/influxdata/telegraf/migrations"
+)
+
+type amqp struct {
+	Headers         map[string]string `toml:"headers"`
+	Database        string            `toml:"database"`
+	RetentionPolicy string            `toml:"retention_policy"`
+
+	Precision string `toml:"precision"`
+
+	Brokers []string `toml:"brokers"`
+	URL     string   `toml:"url"`
+}
+
+func migrate(tbl *ast.Table) ([]byte, string, error) {
+	var old amqp
+	var plugin map[string]interface{}
+	if err := migrations.UnmarshalTableSkipMissing(tbl, &old); err != nil {
+		return nil, "", err
+	}
+	if err := toml.UnmarshalTable(tbl, &plugin); err != nil {
+		return nil, "", err
+	}
+
+	var applied bool
+
+	// Only apply these fields if the headers array is empty, as was the previous behavior.
+	// However, still remove these fields from the toml if they exist
+	doHeaders := len(old.Headers) == 0
+	if old.Database != "" {
+		applied = true
+
+		if doHeaders {
+			if old.Headers == nil {
+				old.Headers = make(map[string]string, 1)
+			}
+			old.Headers["database"] = old.Database
```

Comment:
Please check if the header already exists and conflicts. Return an error in this case. Same below.